lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -3299,13 +3299,59 @@ private function dispatchOnFlushEvent()
         if ($this->evm->hasListeners(Events::onFlush)) {
             $this->evm->dispatchEvent(Events::onFlush, new OnFlushEventArgs($this->em));
         }
+        $subscribedSystems = [];
+        foreach ($this->entityChangeSets as $oid => $entityChangeSet) {
+            foreach ($this->identityMap as $className => $entities) {
+                $hash = (isset($this->entityIdentifiers[$oid]) && isset($this->entityIdentifiers[$oid]['id'])) ? $this->entityIdentifiers[$oid]['id'] : null;
+                if ($hash) {
+                    if (is_object($hash) && method_exists($hash, '__toString')) $hash = $hash->__toString();
+                    if (isset($this->identityMap[$className][$hash])) {
+                        $entity = $this->identityMap[$className][$hash];
+                        if ($oid == spl_object_hash($entity)) {
+                            $entityClass = get_class($entity);
+                            $class = $this->em->getClassMetadata($entityClass);
+                            if (!isset($subscribedSystems[$entityClass])) {
```

Comment:
I'm not entirely sure what this line is violating.  Can you elaborate a bit more?